File: backend/src/main/kotlin/BacklogZero/backend/model/Document.kt
package BacklogZero.backend.model

import jakarta.persistence.*
import java.sql.ResultSet
import java.time.LocalDateTime
import kotlin.jvm.Transient

@Entity
@Table(name = "documents")
data class Document(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long = 0,

        @Column(name = "author_id")
        val authorId: Long,

        @Column(name = "tag_id")
        val tagId: Long,

        @Column(name = "title")
        val title: String,

        @Column(name = "searchable_text")
        val searchableText: String,

        @Column(name = "url")
        val url: String,

        @Column(name = "num_of_upvotes")
        val numOfUpvotes: Int = 0,

        @Column(name = "num_of_downvotes")
        val numOfDownvotes: Int = 0,

        @Column(name = "created_timestamp")
        val createdTimestamp: LocalDateTime,

        @Column(name = "last_updated_timestamp")
        val lastUpdatedTimestamp: LocalDateTime,

        @Transient
        val popularityApprox: Double
)

File: next-js/models/document.js


class Document {
    static _baseUrl = "/documents/";
    static get(id){}

    id;
    author;
    tag;
    root_dummy_comment;
    title;
    searchable_text;
    url;
    num_of_upvotes;
    num_of_downvotes;
    created_timestamp;
    last_updated_timestamp;
    popularity;

    constructor(
        id,
        author_id,
        tag_id,
        root_dummy_comment_id,
        title,
        searchable_text,
        url,
        num_of_upvotes,
        num_of_downvotes,
        created_timestamp,
        last_updated_timestamp,
        popularity
    ){
        this.id = id;
        this.author = ApiGetter(User, author_id);
        this.tag = ApiGetter(Tag, tag_id);
        this.root_dummy_comment = ApiGetter(Comment, root_dummy_comment_id);
        this.title = title;
        this.searchable_text = searchable_text;
        this.url = url;
        this.num_of_upvotes = num_of_upvotes;
        this.num_of_downvotes = num_of_downvotes;
        this.created_timestamp = created_timestamp;
        this.last_updated_timestamp = last_updated_timestamp;
        this.popularity = popularity;
    }
}

